src/unix.rs

```diff
@@ -77,6 +82,32 @@ impl Client {
     }
 
     pub unsafe fn open(s: &str) -> Option<Client> {
+        Client::from_fifo(s).or_else(|| Client::from_pipe(s))
+    }
+
+    /// `--jobserver-auth=fifo:PATH`
+    fn from_fifo(s: &str) -> Option<Client> {
+        let mut parts = s.splitn(2, ':');
+        if parts.next().unwrap() != "fifo" {
+            return None;
+        }
+        let path = match parts.next() {
+            Some(p) => Path::new(p),
+            None => return None,
+        };
+        let file = match OpenOptions::new().read(true).write(true).open(path) {
+            Ok(f) => f,
+            Err(_) => return None,
+        };
+        drop(set_cloexec(file.as_raw_fd(), true));
```

Comment:
I think this can be skipped since `OpenOptions` should always do this for us?

Reply:
You're absolutely right. [std does do that](https://github.com/rust-lang/rust/blob/49b9cc5139dd4d11ef78dc08c1f9170de5b1ca39/library/std/src/sys/unix/fs.rs#L995-L996).